vividus-extension-selenium/src/main/java/org/vividus/ui/variable/AbstractWebDriverDynamicVariable.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ui.variable;
+
+import java.util.Optional;
+
+import org.openqa.selenium.WebDriver;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.variable.DynamicVariable;
+
+public abstract class AbstractWebDriverDynamicVariable implements DynamicVariable
+{
+    private final IWebDriverProvider webDriverProvider;
+
+    protected AbstractWebDriverDynamicVariable(IWebDriverProvider webDriverProvider)
+    {
+        this.webDriverProvider = webDriverProvider;
+    }
+
+    @Override
+    public Optional<String> calculateValue()
+    {
+        if (webDriverProvider.isWebDriverInitialized())
+        {
+            return Optional.of(calculateValue(webDriverProvider.get()));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public String getMissingValueErrorMessage()
+    {
+        return "application is not started";
```

Comment:
```suggestion
        return "session is not started";
```

Reply:
users don't understand what `session` is